moshi/src/main/java/com/squareup/moshi/JsonValueReader.java

```diff
@@ -168,9 +168,20 @@
   }
 
   @Override public int selectString(Options options) throws IOException {
-    String peeked = require(String.class, Token.STRING);
+    Object peeked = (stackSize != 0 ? stack[stackSize - 1] : null);
+    String peekedString;
+
+    if ((peeked instanceof String)) {
```

Comment:
nit: extra parens?

Reply:
or, if you invert this if, `if (!(peeked instance of String)) { ... }` you can avoid separate declaration & assignment for peekedString

Reply:
Yep, thanks!